src/components/DocumentsPage/components/DocumentsEditor/index.js

```diff
@@ -0,0 +1,58 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import './styles.css';
+
+class DocumentsEditor extends Component {
+
+  constructor(props) {
+    super();
+  };
+
+  updateDocument = (event) => {
+    var target = event.target;
+    this.props.onChange(target.name, target.value)
+  };
+
+  delete = () => {
+    if(window.confirm('Delete warning')) {
+      this.props.deleteDocument();
+      this.defaultState();
+    };
+  };
+
+  onSubmit = (event) => {
+    event.preventDefault();
+    this.props.onSubmit(this.props.document);
+  };
+
+  render () {
+    // Ask Kristina why I should do this here :-)
+    var document = {...this.props.document}
```

Comment:
Check if you're properly updating the state (not mutating it) of parent component, when changing - you shouldn't have to do this

Reply:
You don't need to use spread operators here, I removed them locally and everything works:

const document  = this.props.document